src/futures_map.rs

```diff
@@ -127,15 +127,22 @@ where
     /// This iterator returns futures in an arbitrary order, which may change.
     ///
     /// If downcasting a future to `T` fails it will be skipped in the iterator.
-    pub fn iter_of_type<T>(&self) -> impl Iterator<Item = (&ID, &T)>
+    pub fn iter_of_type<T>(&self) -> impl Iterator<Item = (&ID, Pin<&T>)>
     where
         T: 'static,
     {
         self.inner.iter().filter_map(|a| {
             let pin = a.inner.inner.as_ref();
-            let any = Pin::into_inner(pin) as &(dyn Any + Send);
+
+            // Safety: We are only temporarily manipulating the pointer and pinning it again further down.
```

Comment:
This is sound because we're only changing the type of a reference here. If we accessed any of the data or its fields, it could be unsound:
https://doc.rust-lang.org/std/pin/index.html
https://doc.rust-lang.org/std/pin/index.html#subtle-details-and-the-drop-guarantee

```suggestion
            // Safety: We are only changing the type of the pointer, and pinning it again before returning it.
            // None of the fields are accessed.
```